pilot/pkg/model/gateway.go

```diff
@@ -84,12 +85,21 @@ func MergeGateways(gateways ...Config) *MergedGateway {
 
 	log.Debugf("MergeGateways: merging %d gateways", len(gateways))
 	for _, gatewayConfig := range gateways {
-		gatewayName := fmt.Sprintf("%s/%s", gatewayConfig.Namespace, gatewayConfig.Name)
+		gatewayName := gatewayConfig.Namespace + "/" + gatewayConfig.Name // Format: %s/%s
 		names[gatewayName] = true
 
 		gatewayCfg := gatewayConfig.Spec.(*networking.Gateway)
 		log.Debugf("MergeGateways: merging gateway %q into %v:\n%v", gatewayName, names, gatewayCfg)
+		snames := sets.Set{}
 		for _, s := range gatewayCfg.Servers {
+			if len(s.Name) > 0 {
+				if snames.Contains(s.Name) {
+					log.Warnf("Server name %s is not unique in gateway %s and may create possible issues like stat prefix collision ",
```

Comment:
isn't this going to be super spammy? This `name` thing is a new field right? So 100% of users have it unset, meaning that anyone with 2+ gateways will get this every time this codepath is executed, for every gateway, at warn level. I think this code is called pretty often too

Reply:
Should n't this help if len(s.Name) > 0?

Reply:
Yep, thanks! missed that